db.go

```diff
@@ -349,22 +377,31 @@ type DB struct {
 	IgnoreUnmappedCols bool
 	Context            context.Context
 	Logger             QueryLogger
-	mu                 sync.RWMutex
-	models             map[reflect.Type]*Model
-	mappings           map[reflect.Type]fieldMap
+	// Query deadline, ignored if Deadline.IsZero().
+	Deadline time.Time
```

Comment:
Exposing `Deadline` as a public field of `DB` will eventually tempt someone into mutating it outside of `WithContext()`. This might lead to surprising results. (Actually, this applies to `Context` too, but that's out of scope for your change.) I would suggest that you remove the `Deadline` field and just look at the context when a method needs a deadline.